### DISCUSSION TITLE
Accept pure SQL on postgrest, we will be able to use other ORMS on Vercel Edge Runtime

### DISCUSSION THREAD

**Discussion by Jaaneek** [supabase/supabase#14321] [Feature Requests]
Hi,

Since postgrest is a server connected straight to the database, I was wondering if making it accept an optional SQL string would be possible.

I understand that it's not possible on client due to SQL injections. I'm not asking for that.

Not only that, but I would like it to accept SQL only when a service key was provided, meaning it would be only accessible on the server.
This could be very similar to https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript

We would be able to opt out of the bad DX that supabase-js query client gives us and use whatever ORM we want, for example kysely or drizzle are perfect for edge runtime.

The bad DX I'm talking about is mostly typesafety:
https://github.com/supabase/postgrest-js/issues/303

I would also appreciate the ETA on fixing the above issue

**Comment by kouwasi** [comment]
Execute pure sql through postgrest is actually needed?

Supabase is providing a connection pooling with postgres, So you can use it.
Also you can connect without pooling, and can use the prima's pool and others.

https://supabase.com/docs/guides/functions/connect-to-postgres
https://supabase.com/docs/guides/functions/kysely-postgres

**Reply by Jaaneek** [reply]
Seems like this is standard database connection, I want pure HTTP way of sending sql so that I will be able to execute it on vercel/netlify Edge runtime (not deno runtime)

**Reply by kouwasi** [reply]
Do you want exec SQL through HTTP request because your project has some restrictions?

The standard database connection does not depend on runtimes. 
It means that way can Vercel and Netlify runtime.
Same as connecting to a Supabase database from your local machine.

**Reply by Jaaneek** [reply]
Vercel/Netlify Edge runtime only handles HTTP requests, its not possible to connect to a database in there. Other database providers  create special endpoints where you can send the SQL, I would like something similar from supabase

**Reply by kouwasi** [reply]
Oh really?
At first, this request seems weird. but now I 100% agree with your background reason. +1

However, It isn't PostgREST responsibility IMO.
Implement to some other stack?

Anyway, I agree with your request.

**Reply by Jaaneek** [reply]
Yeah, It can be a completely seperate service as far as I'm concerned, PostgREST seems kinda okayish for it because they probably already send SQL into database & have stable database connection. Types would come from our own ORM's (Kysely, drizzle) so not much work required in there, just accept a SQL string and return the database response. 

(It's more complicated that what I described above, if someone is interested here is a more in-depth explanation https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript)

**Reply by Jaaneek** [reply]
Good news! This will be solves by https://github.com/supabase/supavisor do we have ETA on when this will be available with the Supabase?

**Reply by kouwasi** [reply]
Yeah! The supavisor supports multiple protocols, which seems a good solution for your request.
I guess it's available in a few more months from a comment on Hackernews 

> it’s still under development, but it’s at a stage where we can gather a feedback from the community and you can try it yourself. we aren’t using this in production yet, but aiming to deploy it for a subset of databases in the next 2 months.
> 
> https://news.ycombinator.com/item?id=35501718
